mimic/model/nova_objects.py

```diff
@@ -915,6 +915,26 @@ def request_action(self, http_action_request, server_id):
                                          " while it is in task state other than active",
                                          http_action_request))
 
+        elif 'rebuild' in action_json:
+            image_ref = action_json['rebuild'].get('imageRef')
+            if not image_ref:
+                return dumps(bad_request("Could not parse imageRef from request.", http_action_request))
+            if server.status == 'ACTIVE':
+                server.image_ref = image_ref
+                server.status = 'REBUILD'
+                http_action_request.setResponseCode(202)
+                server.collection.clock.callLater(
+                    5.0,
+                    server.update_status,
+                    u"ACTIVE")
+                server_details = server.detail_json(absolutize_url)
+                server_details['adminPass'] = 'password'
+                return dumps({"server": server_details})
+            else:
+                return dumps(conflicting("Cannot 'rebuild' instance " + server_id +
+                                         " while it is in task state other than active",
+                                         http_action_request))
+
```

Comment:
This is a long method that performs a variety of different disparate tasks. It might be worth splitting it into pieces and having the API handler select which method to call.

Nova isn't really my area of expertise, though, so don't let this hold you up if someone else +1's.